packages/config/eslint-preset.js

```diff
@@ -1,40 +1,131 @@
+    sourceType: "module",
+    tsconfigRootDir: __dirname,
+  },
+
+  plugins: ["@typescript-eslint", "import"],
+
+  root: true,
+
   rules: {
+    "@typescript-eslint/array-type": ["error", { default: "array" }],
+    "@typescript-eslint/await-thenable": "error",
+    "@typescript-eslint/consistent-type-imports": ["error", { fixStyle: "inline-type-imports" }],
+    "@typescript-eslint/method-signature-style": "error",
+    "@typescript-eslint/no-meaningless-void-operator": "error",
+    "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
+    "@typescript-eslint/no-non-null-assertion": "error",
+    "@typescript-eslint/no-redundant-type-constituents": "error",
+    "@typescript-eslint/no-unnecessary-condition": "error",
+    "@typescript-eslint/no-unnecessary-type-arguments": "error",
+    "@typescript-eslint/no-unnecessary-type-assertion": "error",
     "@typescript-eslint/no-unused-vars": [
-      "warn",
+      "error",
       {
         argsIgnorePattern: "^_",
-        varsIgnorePattern: "^_",
         caughtErrorsIgnorePattern: "^_",
+        varsIgnorePattern: "^_",
+      },
+    ],
+    "@typescript-eslint/prefer-optional-chain": "error",
+    "@typescript-eslint/prefer-readonly": "error",
+    "@typescript-eslint/promise-function-async": "error",
+    "@typescript-eslint/sort-type-constituents": "error",
+    "@typescript-eslint/switch-exhaustiveness-check": "error",
+    "arrow-body-style": ["error", "as-needed"],
+    "consistent-return": "error",
+    "curly": "error",
+    "eqeqeq": ["error", "smart"],
+    "import/no-duplicates": "error",
+    "import/order": [
+      "error",
+      {
+        groups: ["builtin", "external", "internal", "sibling", "parent"],
+      },
+    ],
+    "no-constant-binary-expression": "error",
+    "no-constructor-return": "error",
+    "no-else-return": "error",
+    "no-implicit-coercion": "error",
+    "no-lonely-if": "error",
+    "no-multi-assign": "error",
+    "no-multi-spaces": "error",
+    "no-new-native-nonconstructor": "error",
+    "no-param-reassign": "error",
+    "no-self-compare": "error",
+    "no-unneeded-ternary": "error",
+    "no-unused-private-class-members": "error",
+    "no-useless-concat": "error",
+    "no-useless-rename": "error",
+    "no-useless-return": "error",
+    "no-var": "error",
+    "object-curly-newline": [
+      "error",
+      {
+        ObjectExpression: {
+          consistent: true,
+          minProperties: 4,
+          multiline: true,
+        },
       },
     ],
+    "object-shorthand": "error",
+    "padding-line-between-statements": [
+      "error",
+      {
+        blankLine: "never",
+        next: "import",
+        prev: "import",
+      },
+    ],
+    "prefer-arrow-callback": "error",
+    "prefer-const": "error",
+    "prefer-template": "error",
+    "quote-props": ["error", "consistent-as-needed"],
+    "react/prop-types": "off",
+    "react/react-in-jsx-scope": "off",
+    "react/display-name": "off",
```

Comment:
This one should not be disabled, it's very useful to have displayNmae when you are debugging forwardRefs etc

Reply:
Take this component: packages/ui/src/components/Icon/Icon.tsx
```js
import { Icon as Iconify } from "@iconify-icon/react"
import { forwardRef } from "react"

export type IconProps = React.ComponentPropsWithoutRef<typeof Iconify>

export const Icon = forwardRef<React.ElementRef<typeof Iconify>, IconProps>((props, ref) => (
  <Iconify ref={ref} {...props} />
))
```

This current implementation has no display name. I understand this makes it hard to debug.

I first thought of fixing it like this:
```js
export const Icon = forwardRef<React.ElementRef<typeof Iconify>, IconProps>(function Icon(props, ref) {
  return <Iconify ref={ref} {...props} />
})
```

However this conflicts with the "prefer-arrow-callback" rule. So doing it like this we should probably disable that rule.

I guess option 2 would be to enforce this pattern? If we want to keep "prefer-arrow-callback"

```js
import { Icon as Iconify } from "@iconify-icon/react"
import { forwardRef } from "react"

export type IconProps = React.ComponentPropsWithoutRef<typeof Iconify>

function IconFn(props: IconProps, ref: React.Ref<React.ElementRef<typeof Iconify>>) {
  return <Iconify ref={ref} {...props} />
}

export const Icon = forwardRef<React.ElementRef<typeof Iconify>, IconProps>(IconFn)
```

Reply:
The recommended way of doing it is with a .displayName accessor

```ts
import { Icon as Iconify } from "@iconify-icon/react"
import { forwardRef } from "react"

export type IconProps = React.ComponentPropsWithoutRef<typeof Iconify>

export const Icon = forwardRef<React.ElementRef<typeof Iconify>, IconProps>((props, ref) => (
  <Iconify ref={ref} {...props} />
))

Icon.displayName = 'Icon'
```

Reply:
Aaa Nice!

Reply:
Fixed in new commits